content/blog/handling-technical-ids-in-gherkin-with-specflow.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "Handling technical id's in Gherkin with SpecFlow"
+date: 2020-06-27T00:00:00+02:00
+image: "images/handling-technical-ids-in-gherkin-with-specflow.jpg"
+tags: [ "Gherkin", "SpecFlow", "Specification by Example", "ATDD", "BDD", "Test Automation", "Cleaner Code" ]
+summary: "Gherkin scenarios in Specification by Example are used to describe the functional requirements of your software. They should be readable for the team and also for the business people that use the software. Technical id's don't have a place here. So what to do when your code requires a technical id?"
+draft: true
+---
+
+When you use Specification by Example with the Gherkin syntax and automate your scenarios with SpecFlow, you're bound to encounter situations where you'll need a technical id. For example to stub data that's retrieved from a repository or external service.
+
+Gherkin scenarios are used to describe the functional requirements of your software. They should be readable for the team and also for the business people that use the software. Technical id's don't have a place in these scenarios. So what to do when your code requires a technical id?
+
+Let start with an example scenario:
+
+```Gherkin
+Given the following people
+    | Id                                   | Address                           |
+    | 9A9EE974-9062-4AB3-98C8-E83B0A5A3BAA | 221B Baker Street, London, UK     |
+    | 70EC5DE6-F569-4092-AF58-DA857F44279E | 1630 Revello Drive, Sunnydale, US |
+    | 0545383F-28E7-4968-9525-11829915ED89 | 31 Spooner Street, Quahog, US     |
+    | EF03C690-6F29-43F0-931F-546938F2869F | 12 Grimmauld Place, London, UK    |
+When person '0545383F-28E7-4968-9525-11829915ED89' moves to '742 Evergreen Terrace, Springfield, US'
+Then the new address of person '0545383F-28E7-4968-9525-11829915ED89' is '742 Evergreen Terrace, Springfield, US'
+```
+
+This scenario describes functionality for moving a person from one address to another.
```

Comment:
Misschien nog een verwijzing naar het probleem? 'Notice how the id does not add anything to the functionality' Dan kan mijn bovenste comment weg.